File: .tmp_py4013/batch_11_dump/3271_hash_divided_string.js
// LeetCode 3271 - Hash Divided String
// https://leetcode.com/problems/hash-divided-string/

var stringHash = function(s, k) {
    let out = '';
    for (let i = 0; i < s.length; i += k) {
        let sum = 0;
        for (let j = i; j < i + k; j++) sum += s.charCodeAt(j) - 97;
        out += String.fromCharCode(97 + sum % 26);
    }
    return out;
};

File: .tmp_rs4013/rust_check15/3271_hash_divided_string.rs
struct Solution;
// LeetCode 3271 - Hash Divided String
// https://leetcode.com/problems/hash-divided-string/

impl Solution {
    pub fn string_hash(s: String, k: i32) -> String {
        let b = s.as_bytes();
        let k = k as usize;
        let mut out = String::new();
        let mut i = 0;
        while i < b.len() {
            let mut sum = 0;
            for j in i..i + k {
                sum += (b[j] - b'a') as i32;
            }
            out.push(char::from(b'a' + (sum % 26) as u8));
            i += k;
        }
        out
    }
}

fn main() {}
